crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -984,9 +984,16 @@ impl CumulocityConverter {
         match &channel {
             Channel::EntityMetadata => {
                 if let Ok(register_message) = EntityRegistrationMessage::try_from(message) {
-                    return self
-                        .try_register_entity_with_pending_children(register_message)
-                        .await;
+                    let mut converted_messages = self
+                        .try_register_entity_with_pending_children(&register_message)
+                        .await?;
+
+                    // Republish the metadata message with @id if it's not given
+                    let mut updated_metadata_message =
+                        self.append_id_if_not_given(register_message, &source);
+                    converted_messages.append(&mut updated_metadata_message);
```

Comment:
You may have to push this logic into the `try_register_entity_with_pending_children` function so that the same logic is applied to the pending children as well. They should also have their generated `@id`s published. Once that's done, add some test coverage as well for the pending entities. Refer to `registration_lifecycle.robot::Early data messages cached and processed` for some samples.

Reply:
Addressed in 0ef0227031078fd9e74e6073437fde2ea113bd72